demos/burgers-isotropic.py

```diff
@@ -0,0 +1,274 @@
+
+            # append the metric for the step in the time partition
+            _metrics.append(_metric)
+
+        # set the first metric to the base
+        metric = _metrics[0]
+        # all the other metrics
+        metric.average(*_metrics[1:], weights=[dt] * len(_metrics))
+
+        metrics.append(metric)
+
+    # Apply space time normalisation
+    space_time_normalise(metrics, mesh_seq.time_partition, mp)
+
+    # Adapt each mesh w.r.t. the corresponding metric, provided it hasn't converged
+    for i, metric in enumerate(metrics):
+        if not mesh_seq.converged[i]:
+            mesh_seq[i] = adapt(mesh_seq[i], metric)
+        complexities.append(metric.complexity())
+    num_dofs = mesh_seq.count_vertices()
+    num_elem = mesh_seq.count_elements()
+    pyrint(f"fixed point iteration {iteration + 1}:")
+    for i, (complexity, ndofs, nelem) in enumerate(
+        zip(complexities, num_dofs, num_elem)
+    ):
+        pyrint(
+            f"  subinterval {i}, complexity: {complexity:4.0f}"
+            f", dofs: {ndofs:4d}, elements: {nelem:4d}"
+        )
+
+    # Plot each intermediate adapted mesh
+    fig, axes = mesh_seq.plot()
+    for i, ax in enumerate(axes):
+        ax.set_title(f"Subinterval {i + 1}")
+    fig.savefig(f"burgers-hessian_mesh{iteration + 1}.jpg")
+    plt.close()
+
+    # Since we have two subintervals, we should check if the target complexity has been
+    # (approximately) reached on each subinterval
+    continue_unconditionally = np.array(complexities) < 0.90 * target
+    return continue_unconditionally
+
+
+# With the adaptor function defined, we can call the fixed point iteration method. ::
+
+solutions = mesh_seq.fixed_point_iteration(
+    adaptor,
+    enrichment_kwargs={
+        "enrichment_method": "h",
+    },
+)
+
+# Here the output should look something like
+#
+# .. code-block:: console
+#
+# fixed point iteration 1:
+#   subinterval 0, complexity:  449, dofs:  619, elements: 1162
+#   subinterval 1, complexity:  350, dofs:  451, elements:  833
+# fixed point iteration 2:
+#   subinterval 0, complexity:  654, dofs:  778, elements: 1453
+#   subinterval 1, complexity:  539, dofs:  643, elements: 1199
+# fixed point iteration 3:
+#   subinterval 0, complexity:  884, dofs:  945, elements: 1789
+#   subinterval 1, complexity:  709, dofs:  795, elements: 1487
+# QoI converged after 4 iterations under relative tolerance 0.001.
+
+# Finally, let's plot the adapted meshes. ::
+
+fig, axes = mesh_seq.plot()
+for i, ax in enumerate(axes):
+    ax.set_title(f"Subinterval {i + 1}")
+fig.savefig("burgers-go_iso_mesh.jpg")
+plt.close()
+
+# .. figure:: burgers-iso_mesh.jpg
+#    :figwidth: 100%
+#    :align: center
+#
+# Recall that the Burgers problem is quasi-1D, since the analytical solution varies in
+# the :math:`x`-direction, but is constant in the :math:`y`-direction. As such, we can
+# affort to have lower resolution in the :math:`y`-direction in adapted meshes. That
```

Comment:
I find this a bit misleading. To me, having lower resolution in the y-direction would mean anisotropy. But these meshes are isotropic. Did I misunderstand something?

Reply:
@jwallwork23  See https://github.com/mesh-adaptation/goalie/commit/6dba2b83e6185981ec8af509142809b68bc2fb81: I edit the text to be consistent with the [goal-based point-discharge demo](https://github.com/mesh-adaptation/goalie/blob/main/demos/point_discharge2d-goal_oriented.py) and similarly extended the demo to include an anisotropic goal-based adaptor function.